nixos/modules/services/network-filesystems/samba.nix

```
@@ -191,6 +176,100 @@ in
         networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ 139 445 ];
         networking.firewall.allowedUDPPorts = mkIf cfg.openFirewall [ 137 138 ];
       })
-    ];
 
+      (lib.mkIf cfg.nmbd.enable {
+        systemd.services.samba-nmbd = {
+          description = "Samba NMB Daemon";
+          documentation = [ "man:nmbd(8)" "man:samba(7)" "man:smb.conf(5)" ];
+
+          after = [
+            "network.target"
+            "network-online.target"
+          ];
+
+          partOf = [ "samba.target" ];
+          wantedBy = [ "samba.target" ];
+          wants = [ "network-online.target" ];
+
+          environment.LD_LIBRARY_PATH = config.system.nssModules.path;
+
+          serviceConfig = {
+            ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+            ExecStart = "${cfg.package}/sbin/nmbd --foreground --no-process-group ${lib.escapeShellArgs cfg.nmbd.extraArgs}";
+            LimitCORE = "infinity";
+            PIDFile = "/run/nmbd.pid";
```

```suggestion
            PIDFile = "/run/samba/nmbd.pid";
```
We should also try to add some hardening